### ISSUE TITLE
Way to disable the warnings for missing aria-label on md-buttons

### ISSUE THREAD

**Issue by izeau** [angular/material#3441]
Hi,

I'm using angular-translate for internationalization, so my HTML looks like this:

```html
<md-button translate="key"></md-button>
```

Now, I get why these warnings exist, but they're clogging my console and there doesn't seem to be a way to disable them.

Duplicating code is annoying, so I really don't want to `<md-button translate="key" aria-label="{{ 'key' | translate }}></md-button>`. I know I could use `<md-button translate>key</md-button>` but let's be honest here, it's just a workaround and requires cloaking to not display the key to the end-user.

Anyway, if there were a way to disable these warnings it would be nice.

Thanks!

**Comment by marcysutton** [comment]
Can you post a demo showing the code in action? It would be better to add support for Angular translate, rather than add a back door to disable warnings. Perhaps it's a timing issue that just needs to be resolved.

**Comment by izeau** [comment]
Here you go: http://codepen.io/anon/pen/qdpdPy

Yes, it could probably be fixed by tweaking the priority of the directives, but imho we should be able to disable library warnings if we want to, especially when said warning are shown to the end user.

Warning comes from angular material, it is not angular-translate's job to add support for material or material's job to add support for the `translate` directive.

**Comment by izeau** [comment]
In case anyone has the same issue, I wrote this simple `translate` directive / workaround, that auto-injects the `aria-label` attribute during the compile phase.

```js
module.directive('translate', function ($interpolate) {
  return {
    priority: 10,
    compile: compile
  };

  function compile (tElement, tAttr) {
    var translate   = tAttr.translate;
    var trValues    = tAttr.translateValues;
    var ariaLabel   = [
      $interpolate.startSymbol(),
      JSON.stringify(translate),
      ' | translate',
      trValues ? (':' + trValues + ' ') : '',
      $interpolate.endSymbol()
    ].join('');
    var interpolate = $interpolate(ariaLabel);

    tAttr.$set('ariaLabel', ariaLabel);

    return function postLink (scope, element, attr) {
      attr.$set('ariaLabel', interpolate(scope));
    };
  }
});
```

**Comment by marcysutton** [comment]
Nice! One note: the attribute needs a hyphen to be valid and usable. So `ariaLabel` has to become `'aria-label'` as a string.

**Comment by super-coder** [comment]
+1

**Comment by kevinlbatchelor** [comment]
+1

**Comment by thanuzz** [comment]
+1

**Comment by jpolo** [comment]
Similar problem for me 
```html
<!-- angular translate is needed -->
<md-button class="md-icon-button"
             translate 
             translate-attr-aria-label="navigation_action_search_close"
             ng-click="spNavigationBarCrud.setSearchBarVisible(false)">
    <md-icon>close</md-icon>
  </md-button>
```

It raises a warning although the aria-label was set. MD might check too early the presence of aria-label and does not allow any other directive to set it...

**Comment by yanndupuy** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  dupuyyann@gmail.com
    Domain zyapps.com has exceeded the max emails per hour (429/400 (107%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext5.iad.github.net ([192.30.252.196]:39690 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by serveur.ex2-p10.com with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.87)
	(envelope-from <noreply@github.com>)
	id 1avsZn-001Z8Q-5W
	for yann.dupuy@zyapps.com; Thu, 28 Apr 2016 22:30:20 +0200
Date: Thu, 28 Apr 2016 13:27:50 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1461875270;
	bh=cn1adS/CtQXCGDpW52vfLR1n3EDhQK+pAFiW3JAQd1Y=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=LLlKglrDvPWE124yLU+V9oF0JDnHRoHNjQNDhIcABVxt4rEomScrdNk65PSkQeWJw
	 7RScB1buPVJdWs6i607Ph01pLWSL6ZFogGFHSiNOjW4H2FGuC2E0rH+UllkzzCl3SZ
	 iYT8hWMPT/mS/rcjdDZ1fwOXwcbzahTc2zR7og6E=

**Comment by 1mike12** [comment]
This is the most annoying thing. I'm working on something that I will be personally using. I don't want a million aria warnings in the console. I completely understand the standardization you're trying to enforce to make the web more accessible, but to not ever be able to turn it off, even when you're trying to hack something quickly or working on something the public isn't going to see makes no sense.

**Comment by sinedied** [comment]
@1mike12 +1

This should be at least configurable, as not everyone wishes to add full ARIA support to their app.
Even then, not using the aria-label on an HTML button object, most (if not all) screen readers will fall back to reading the button text instead, so it does not hurts accessibility at all, just the devs...

**Comment by samal84** [comment]
@jpolo At least it seems that Angular translate will overwrite `aria-label` if you provide `translate-attr-aria-label`, so you can just put in a placeholder `aria-label` to supress the warning.

<md-button class="md-icon-button"
             aria-label="this will get overwritten"
             translate 
             translate-attr-aria-label="navigation_action_search_close"
             ng-click="spNavigationBarCrud.setSearchBarVisible(false)">
    <md-icon>close</md-icon>
  </md-button>

**Comment by marcysutton** [comment]
That's a lot of effort to avoid putting a label on something! Keep in mind some elements need the `aria-label` to expose an accessible name (like `md-radio-button`), but `md-button` is accessible with plain text. If you're getting a warning thrown when there is text in mdButton, it's a bug and should be filed. 

That said, you'll be happy to hear they did add a method to disable the warnings: https://github.com/angular/material/commit/d63e4d0ae9b2236b5de2f8f5e7b81865e5338ee1

**Comment by stephengardner** [comment]
```js
myApp.config(['$mdAriaProvider', function ($mdAriaProvider) {
		$mdAriaProvider.disableWarnings();
	}])
```

Ta-da...

**Comment by curtwagner1984** [comment]
@stephengardner Bless you god Sir!

**Comment by Jaya1993** [comment]
$mdAriaProvider.disableWarnings(); is providing me with an error -> Uncaught Error: [$injector:modulerr]
